packages/react-dom/src/__tests__/findDOMNode-test.js

```diff
@@ -163,4 +163,26 @@ describe('findDOMNode', () => {
     ]);
     expect(match).toBe(child);
   });
+
+  it('findDOMNode should not warn if passed a host component inside StrictMode', () => {
+    let child = undefined;
+
+    class IsInStrictMode extends React.Component {
+      render() {
+        const {as: Component} = this.props;
+        return <Component ref={n => (child = n)} />;
+      }
+    }
+
+    ReactTestUtils.renderIntoDocument(
+      <StrictMode>
+        <IsInStrictMode as="span" />
+      </StrictMode>,
+    );
+
+    // outside of dev toWarnDev would always pass which means negating it would always fail
+    if (__DEV__) {
+      expect(() => ReactDOM.findDOMNode(child)).not.toWarnDev(['**']);
```

Comment:
Would this also match a potential future warning? I had the full message originally but I'm not sure how it would actually look like
```suggestion
      expect(() => ReactDOM.findDOMNode(child)).not.toWarnDev(['Warning: findDOMNode is deprecated in StrictMode.**']);
```